### PR TITLE
DC-597: add sonar scan to circleci build process

### PR DESCRIPTION
**Pull request by pshapiro4broad**
This should produce a circle ci scan that includes coverage data.

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+# This is the name and version displayed in the SonarCloud UI.
+#sonar.projectName=terra-ui
+#sonar.projectVersion=1.0
+
+# Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
+#sonar.sources=.
+
+# Encoding of the source code. Default is default system encoding
+#sonar.sourceEncoding=UTF-8
```
Comment (kyuksel) [inline sonar-project.properties]: Should we remove the commented-out portion?

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,13 @@
+sonar.projectKey=DataBiosphere_terra-ui
+sonar.organization=broad-databiosphere
+
+# This is the name and version displayed in the SonarCloud UI.
```
Comment (nawatts) [inline sonar-project.properties]: Could these commented out properties be removed?

**Review by nawatts — APPROVED** [review]
Thanks for doing this!

It looks like this doesn't yet include the coverage report in SonarCloud though (https://github.com/DataBiosphere/terra-ui/pull/3620#issuecomment-1362152784 says "No Coverage Information").

It looks like coverage requires setting another property in `sonar-project.properties`: https://docs.sonarcloud.io/enriching/test-coverage/javascript-typescript-test-coverage/#adding-the-coverage-analysis-parameter

**Review by TomConner — APPROVED** [review]

**Comment by nawatts** [comment]
Nice. Though the [comment from sonarcloud](https://github.com/DataBiosphere/terra-ui/pull/3620#issuecomment-1362178845) here still says "No coverage information", it looks like it did pick coverage. Compare the coverage tile for this PR vs a previous one:
- https://sonarcloud.io/summary/new_code?id=DataBiosphere_terra-ui&pullRequest=3620
- https://sonarcloud.io/summary/new_code?id=DataBiosphere_terra-ui&pullRequest=3600

Maybe it says "No coverage information" because this PR didn't change any JS code?

**Pull request merged**